src/ASM/ASMs2Dmx.C

```diff
@@ -222,7 +222,10 @@ bool ASMs2Dmx::generateFEMTopology ()
       return false; // Logic error
   }
   delete surf;
-  geomB = surf = m_basis[elmBasis-1]->clone();
+  const Go::SplineSurface* surfOld = surf;
```

Comment:
Maybe a `bool geoAndItgBasisAreSame = surf == geomB;` instead of copying the pointer would be more readable? That variable could also be a class member as I assume it will be rather constant (but the name is up for discussion).

Or can `ASMbase::refineGeometry` be used instead?